src/main/java/com/elixr/poc/bulkimport/service/reader/FileReader.java

```diff
@@ -0,0 +1,165 @@
+package com.elixr.poc.bulkimport.service.reader;
+
+import com.elixr.poc.bulkimport.dto.Patient;
+
+import com.elixr.poc.bulkimport.response.GenericResponse;
+import com.elixr.poc.bulkimport.response.RowResponse;
+import com.elixr.poc.bulkimport.service.FileOperationService;
+import com.elixr.poc.common.enums.FileOperationEnum;
+import com.elixr.poc.common.enums.MessagesKeyEnum;
+import com.elixr.poc.common.util.MessagesUtil;
+import io.micrometer.common.util.StringUtils;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellType;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.xssf.usermodel.XSSFRow;
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.*;
+
+/**
+ * Service layer for fileUpload operation.
+ */
+@Slf4j
+@Service
+public class FileReader {
+    private final FileOperationService fileOperationService;
+
+    public FileReader(FileOperationService fileOperationService) {
+        this.fileOperationService = fileOperationService;
+    }
+
+    String stringValue = null;
+
+    /**
+     * Reading the Excel file and performing the Operation based on Action.
+     *
+     * @param multipartFile
+     * @return
+     * @throws IOException
+     */
+    public GenericResponse readData(MultipartFile multipartFile) throws IOException {
+        InputStream inputStream = multipartFile.getInputStream();
+        String[] patientKey = {FileOperationEnum.ACTION.getFileKey(), FileOperationEnum.PATIENT_ID.getFileKey(),
+                FileOperationEnum.PATIENT_FIRSTNAME.getFileKey(), FileOperationEnum.PATIENT_LASTNAME.getFileKey(),
+                FileOperationEnum.PATIENT_AGE.getFileKey(), FileOperationEnum.PATIENT_ADDRESS.getFileKey(),
+                FileOperationEnum.DOCTOR_ID.getFileKey()};
+        HashMap<String, String> columnHeader = new HashMap<>();
+        try {
+            XSSFWorkbook workbook = new XSSFWorkbook(inputStream);
+            XSSFSheet sheet = workbook.getSheetAt(0);
+            Row row;
+            Cell cell;
+            List<RowResponse> rowResponse = new ArrayList<>();
+            Iterator rowIterator = sheet.rowIterator();
+            while (rowIterator.hasNext()) {
+                row = (XSSFRow) rowIterator.next();
+                if (row.getRowNum() != 0) {
+                    int headerCount = 0;
```

Comment:
Name it as currentHeaderIndex. HeaderCount gives impression that it is total count of headers in the excel

Reply:
Updated.